lib/kcerts/ssh.go

```diff
@@ -141,15 +141,15 @@ func (a *SSHAgent) UseStandardPaths() error {
 	tempname := fmt.Sprintf("%s/enkit.tmp%016x", path, mathrand.New(srand.Source).Uint64())
 	defer os.Remove(tempname)
 	if err := os.Symlink(a.Socket, tempname); err != nil {
-		return err
+		return fmt.Errorf("UseStandardPaths symlink failed: %w", err)
 	}
 	// Rename symlink to the standard name
-	os.Remove(socket) // if it exists
-	if err := os.Rename(tempname, socket); err != nil {
-		return err
+	os.Remove(standard_socket_path) // if it exists
```

Comment:
remove the Remove? os.Rename will overwrite/remove the target on its own if it exists.
Even better: it will do so atomically (on most posix filesystems), so there is no window during which the link may not exist.

See: https://pkg.go.dev/os#Rename, and documentation around the rename syscall.